modules/pam_faillock/faillock.c

```diff
@@ -76,13 +76,22 @@ open_tally (const char *dir, const char *user, uid_t uid, int create)
 		flags |= O_CREAT;
 	}
 
-	fd = open(path, flags, 0600);
+	fd = open(path, flags, 0660);
 
 	free(path);
 
 	if (fd != -1) {
 		struct stat st;
 
+		/*
+		 * If umask is set to 022, as will probably in most systems, then the
+		 * group will not be able to write to the file. So, change the file
+		 * permissions just in case.
+		 */
+		if (fchmod(fd, 0660)) {
```

Comment:
As this is auxiliary call I'd suggest using ignore_return() instead of returning error.
Perhaps also add note to the comment that the user will be set to the real user uid and the group would be root so later root processes will require CAP_DAC_OVERRIDE.